clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -74,8 +74,22 @@ public KafkaPrincipal principal() {
     public void prepare() throws IOException {
         if (!transportLayer.ready())
             transportLayer.handshake();
-        if (transportLayer.ready() && !authenticator.complete())
-            authenticator.authenticate();
+        if (transportLayer.ready() && !authenticator.complete()) {
+            try {
+                authenticator.authenticate();
+            } catch (AuthenticationException e) {
+                switch (authenticator.error()) {
+                    case AUTHENTICATION_FAILED:
+                    case ILLEGAL_SASL_STATE:
+                    case UNSUPPORTED_SASL_MECHANISM:
```

Comment:
UNSUPPORTED_SASL_MECHANISM error can be mapped to UNSUPPORTED_AUTHENTICATION_MECHANISM state?

Reply:
@omkreddy Thanks for the review. I have removed the unused state `UNSUPPORTED_AUTHENTICATION_MECHANISM`. It was added earlier to enable clients to log a different error message for this case But now, brokers return an error message in their SaslAuthenticate response which is propagated to clients, so we no longer need that state.